### ISSUE TITLE
[CI[ SearchScrollIT#testCloseAndReopenOrDeleteWithActiveScroll failure

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#25465]
```
REPRODUCE WITH: gradle :core:integTest -Dtests.seed=825FE78E1F11CB4F -Dtests.class=org.elasticsearch.search.scroll.SearchScrollIT -Dtests.method="testCloseAndReopenOrDeleteWithActiveScroll" -Dtests.security.manager=true -Dtests.locale=es-CU -Dtests.timezone=America/Montreal
   > Throwable #1: java.lang.AssertionError: timed out waiting for green state
   > 	at org.elasticsearch.test.ESIntegTestCase.ensureColor(ESIntegTestCase.java:944)
   > 	at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:910)
   > 	at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:899)
   > 	at org.elasticsearch.search.scroll.SearchScrollIT.testCloseAndReopenOrDeleteWithActiveScroll(SearchScrollIT.java:515)
   > 	at java.lang.Thread.run(Thread.java:748)Throwable #2: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:722)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:696)
   > 	at org.elasticsearch.test.InternalTestCluster.assertRequestsFinished(InternalTestCluster.java:2069)
   > 	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2048)
   > 	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:586)
   > 	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2067)
   > 	at java.lang.Thread.run(Thread.java:748)
```

Doesn't reproduce.

Build url: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=debian/1183/consoleText

**Comment by martijnvg** [comment]
This failure looks related: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+g1gc/3350/console

```
FAILURE 10.5s J0 | SearchScrollIT.testSimpleScrollQueryThenFetchClearAllScrollIds <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 	at __randomizedtesting.SeedInfo.seed([40E655E03F96863C:1A3F3D70AB1DC4DC]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:722)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:696)
   > 	at org.elasticsearch.test.InternalTestCluster.assertRequestsFinished(InternalTestCluster.java:2069)
   > 	at org.elasticsearch.test.InternalTestCluster.assertAfterTest(InternalTestCluster.java:2048)
   > 	at org.elasticsearch.test.ESIntegTestCase.afterInternal(ESIntegTestCase.java:586)
   > 	at org.elasticsearch.test.ESIntegTestCase.cleanUpCluster(ESIntegTestCase.java:2067)
   > 	at java.lang.Thread.run(Thread.java:748)
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
  2> REPRODUCE WITH: gradle :core:integTest -Dtests.seed=40E655E03F96863C -Dtests.class=org.elasticsearch.search.scroll.SearchScrollIT -Dtests.method="testSimpleScrollQueryThenFetch_clearScrollIds" -Dtests.security.manager=true -Dtests.jvm.argline="-XX:-UseConcMarkSweepGC -XX:+UseG1GC" -Dtests.locale=hi -Dtests.timezone=America/Boa_Vista
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
   > 	Suppressed: java.lang.AssertionError: All incoming requests on node [node_sd3] should have finished. Expected 0 but got 182
   > Expected: <0L>
   >      but: was <182L>
   > 		at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 		at org.elasticsearch.test.InternalTestCluster.lambda$assertRequestsFinished$17(InternalTestCluster.java:2072)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 39 more
```

**Comment by markharwood** [comment]
Another one, didn't reproduce on my laptop:

    gradle :core:integTest -Dtests.seed=4612B379B8EDEA93 -Dtests.class=org.elasticsearch.search.scroll.SearchScrollIT -Dtests.method="testStringSortMissingAscTerminates" -Dtests.security.manager=true -Dtests.locale=zh-TW -Dtests.timezone=US/Alaska

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-windows-compatibility/398/console

	 1> [2017-07-03T07:08:17,318][INFO ][o.e.s.s.SearchScrollIT   ] [testCloseAndReopenOrDeleteWithActiveScroll]: after test

	ERROR   41.2s J0 | SearchScrollIT.testCloseAndReopenOrDeleteWithActiveScroll <<< FAILURES!
	   > Throwable #1: java.lang.AssertionError: timed out waiting for green state
	   > 	at org.elasticsearch.test.ESIntegTestCase.ensureColor(ESIntegTestCase.java:944)
	   > 	at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:910)
	   > 	at org.elasticsearch.test.ESIntegTestCase.ensureGreen(ESIntegTestCase.java:899)
	   > 	at org.elasticsearch.search.scroll.SearchScrollIT.testCloseAndReopenOrDeleteWithActiveScroll(SearchScrollIT.java:515)
	   > 	at java.lang.Thread.run(Thread.java:745)Throwable #2: java.lang.AssertionError: All incoming requests on node [node_sd5] should have finished. Expected 0 but got 182
	   > Expected: <0L>
	   >      but: was <182L>
	   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)

**Comment by markharwood** [comment]
And another non-reproducing one with the same error:

    gradle :core:integTest -Dtests.seed=3AA4D4E1475D5334 -Dtests.class=org.elasticsearch.search.scroll.SearchScrollIT -Dtests.method="ARSEtestCloseAndReopenOrDeleteWithActiveScroll" -Dtests.security.manager=true -Dtests.locale=lv-LV -Dtests.timezone=Europe/Malta

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+multijob-unix-compatibility/os=debian/1200/console

**Comment by jimczi** [comment]
All these failures seems to stem from the fact that the global checkpoint not being in sync with the local checkpoint at startup of one node:

````
[org.elasticsearch.transport.AssertingTransportInterceptor$TestPlugin]
  2> jūl. 03, 2017 5:11:44 PM com.carrotsearch.randomizedtesting.RandomizedRunner$QueueUncaughtExceptionsHandler uncaughtException
  1> [2017-07-03T17:11:42,812][INFO ][o.e.p.PluginsService     ] [node_s1] loaded plugin [org.elasticsearch.transport.MockTcpTransportPlugin]
  1> [2017-07-03T17:11:42,823][INFO ][o.e.d.DiscoveryModule    ] [node_s1] using discovery type [test-zen]
  2> WARNING: Uncaught exception in thread: Thread[elasticsearch[node_s1][generic][T#1],5,TGRP-SearchScrollIT]
  2> java.lang.AssertionError: local checkpoint [-2] for allocation ID [duZojjjjRMavMOLvPQbT5A] violates being at least the global checkpoint [8]
  2> 	at __randomizedtesting.SeedInfo.seed([3AA4D4E1475D5334]:0)
````

All failures have this assertion error in their logs. @jasontedor I am reassigned this to you, feel free to cancel the reassignment but this does not seem to be caused by scroll queries.

**Comment by jasontedor** [comment]
Duplicates #25415, closed by #25468
